File: frontend/src/types/Asteroid.ts
export interface Asteroid {
    id: string;
    name: string; 
    is_potentially_hazardous_asteroid: boolean;
    close_approach_data: CloseApproachData[];
}

export interface AsteroidDetails {
    id: string;
    neo_reference_id: string;
    name: string; 
    designation: string;
    nasa_jpl_url: string;
    absolute_magnitude_h: number;
    is_potentially_hazardous_asteroid: boolean;
}

interface CloseApproachData {
    close_approach_date: string;
    miss_distance: MissDistance;

}
interface MissDistance {
    kilometers: string;
}

File: src/main/java/org/nasa/neo/dto/Asteroid.java
package org.nasa.neo.dto;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.time.LocalDate;
import java.util.List;

@JsonIgnoreProperties(ignoreUnknown = true)
public record Asteroid(
        String id,
        String name,
        @JsonProperty("is_potentially_hazardous_asteroid")
        boolean isPotentiallyHazardous,
        @JsonProperty("close_approach_data")
        List<CloseApproachData> closeApproachData
) {
        public record CloseApproachData(
                @JsonProperty("miss_distance")
                MissDistance missDistance,
                @JsonProperty("close_approach_date")
                @JsonFormat(pattern="yyyy-MM-dd")
                LocalDate approachDate
        ) { }

        public record MissDistance(@JsonProperty("kilometers") String kilometers) {
                public double missDistanceInKilometers() {
                        return Double.parseDouble(kilometers);
                }
        }
}
